### ISSUE TITLE
Checker framework doesn't output similar logs for Windows and Linux

### ISSUE THREAD

**Issue by Vyom-Yadav** [typetools/checker-framework#5665]
We have started to integrate Checker Framework with the Checkstyle project.

The logs in Windows and Linux are not similar.

From Windows: (snippet from [complete logs](https://gist.githubusercontent.com/Vyom-Yadav/6552bbcf8b691721eda65b8207e5714c/raw/087ccf374b34a2f58e9761049f4cce881a33025a/log.txt))
```
[WARNING] D:\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\TranslationCheck.java:[632,16] [argument] incompatible argument for parameter args of Violation.
  found   : @Initialized @NonNull String @Initialized @Nullable []
[WARNING]...........
```

From Linux: (snippet from [complete logs](https://gist.githubusercontent.com/Vyom-Yadav/2f19bf300300a8b23b1c5b86e660ab07/raw/eb6eed215e9a0652a3a99a0722046bec834dfd08/log-linux.txt))
```
[WARNING] /home/vyom/IdeaProjects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:[342,20] [argument] incompatible argument for parameter args of Violation.
  found   : @Initialized @Nullable String @Initialized @NonNull []
  required: @Initialized @NonNull Object @Initialized @NonNull []
[WARNING]...........
```

Along with this the order of logs is also incorrect due to a [known issue](https://issues.apache.org/jira/browse/MCOMPILER-509?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel).

The string `required: @Initialized @NonNull Object @Initialized @NonNull []` isn't present anywhere in Windows logs. 

## How to reproduce the issue:

Clone the repo https://github.com/checkstyle/checkstyle

```bash
$ mvn -e --no-transfer-progress clean compile -P"checker-nullness-optional-interning",no-validations
```

The issue in Checkstyle is being tracked at https://github.com/checkstyle/checkstyle/issues/12695

---
Linux:
```bash
$ java --version                                                                                    
openjdk 17.0.6 2023-01-17
OpenJDK Runtime Environment (Red_Hat-17.0.6.0.10-1.fc37) (build 17.0.6+10)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.6.0.10-1.fc37) (build 17.0.6+10, mixed mode, sharing)
```

Windows:
```cmd

```

**Comment by Vyom-Yadav** [comment]
@msridhar @mernst Can you please triage this issue?

**Comment by Vyom-Yadav** [comment]
Ideally, resolving https://github.com/typetools/checker-framework/issues/5666 would be the best alongside this issue. I am up for helping build a report of the analysis CF does.

**Comment by mernst** [comment]
Thanks for the ping on this.  That bug on Windows is a shame!

We have been busy with a number of things, so this wasn't at the top of our list.  However, since you are willing to help (thanks!), then let's get this done.

What assistance do you need from us in order to get started?

**Comment by Vyom-Yadav** [comment]
@mernst I just want to confirm that the problem in the Windows logs is due to some logging problem (buffer full, some logs dropped, etc.), and CF is producing the correct results. Just writing to some output stream is buggy, right?

In that case, I can help in making a report about the errors given by CF and maybe fix the logging problem too.

**Comment by mernst** [comment]
I misinterpreted your text "Along with this the order of logs is also incorrect due to a known issue" as saying that both the dropped text and the order of logs are due to a known issue.  Now I see that you are saying these are distinct, unrelated issues.  So, let's ignore the order of logs and focus on the missing text.

I'm not a Windows user.  I haven't previously heard this complaint from other users.  @markro have you experienced any missing log text, as in this bug report, when running the Checker Framework?

Since I can't reproduce your problem, I really appreciate your help in diagnosing and/or fixing the problem.

This message is being produced by the template [`argument`](https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/common/basetype/messages.properties#L19).  That template contains two `%n` directives that ought to be properly interpreted on Windows.  What's more, they ought to be interpreted the same, yet only the last line is getting lost.

I have seen before that some tools swallow all but the first line of an error message, but I haven't seen a tool swallow just some of the subsequent lines.

Please try adding `%n` to the very end of that message to see whether that makes a difference.  I'm not optimistic about this, but it is worth trying to investigate whether just the last line is getting lost.

**Comment by romani** [comment]
@mernst , is it possible to create report file from validation? and not rely on any output of plugin.
We use output only because there is no report file of any format that we can use to show validation in more exact form.

**Comment by romani** [comment]
I would say, #5666 will allow users to have workaround.

But issue with broken output is different issue, somebody else might run into it also if they start to deal with violations from this tool.
